src/app/active-mode/Views/CurrentActivity.tsx

```diff
@@ -31,56 +33,105 @@ function mapStateToProps(state: RootState): StoreProps {
 
 type Props = ProvidedProps & StoreProps;
 
+const ACTIVITY_REFRESH_RATE = 10_000;
+let cooldown = Date.now();
+let intervalId = 0;
+let firstLoad = true;
+
 async function refreshActivity({ account, store }: { account: DestinyAccount; store: DimStore }) {
   const profileInfo = await getCurrentActivity(account);
-  return profileInfo.characterActivities.data?.[store.id].currentActivityHash;
+  return profileInfo.characterActivities.data?.[store.id];
 }
 
 function CurrentActivity({ account, store, defs, buckets }: Props) {
-  const [hash, setHash] = useState<number | undefined>();
+  const [activityInfo, setActivityInfo] = useState<
+    DestinyCharacterActivitiesComponent | undefined
+  >();
+  const prevActivityInfo = useRef<DestinyCharacterActivitiesComponent | undefined>();
+
   useEffect(() => {
-    refreshActivity({ account, store }).then(setHash);
-  }, [defs, account, store]);
+    if (firstLoad) {
+      refreshActivity({ account, store }).then(setActivityInfo);
+    }
 
-  if (!defs || !hash || hash === Destinations.Orbit) {
-    return null;
-  }
+    window.clearInterval(intervalId);
+    intervalId = window.setInterval(() => {
+      // If the activity mode just changed, trust the new one and don't ping for a little.
+      // Sometimes there is a weird cache issue, where even after some time it flips back and forth with the last
+      if (Date.now() - cooldown < ACTIVITY_REFRESH_RATE * 6) {
+        return;
+      }
+
+      refreshActivity({ account, store }).then(setActivityInfo);
+    }, ACTIVITY_REFRESH_RATE);
 
-  const activity = defs.Activity.get(hash);
+    return () => {
+      window.clearInterval(intervalId);
+      firstLoad = true;
+    };
+  }, [account, store]);
 
-  if (!activity) {
+  useEffect(() => {
+    if (!activityInfo) {
+      return;
+    }
+    if (
+      !firstLoad &&
+      prevActivityInfo.current?.currentActivityHash !== activityInfo.currentActivityHash
+    ) {
+      refresh();
+      cooldown = Date.now();
+    }
+    firstLoad = false;
+    prevActivityInfo.current = activityInfo;
+  }, [activityInfo]);
```

Comment:
This feels like a good candidate for extracting into a `useActivityInfo` hook

Reply:
I moved it to its own function, in the util file as well, seems to be working at first glance, but it may need another look